src/main/java/com/devonfw/ide/sonarqube/common/impl/check/DevonArchitectureCheck.java

```diff
@@ -96,7 +96,10 @@ public void visitNewClass(NewClassTree newClassTree) {
   @Override
   public void visitClass(ClassTree classTree) {
 
-    String simpleName = classTree.simpleName().name();
+    String simpleName;
+    if ((simpleName = classTree.simpleName().name()) == null) {

```

Comment:
The problem was that `simpleName()` returned `null`.
So you need to extract `classTree.simpleName()` to a local variable and check that one for null.
Sorry, in case I was not clear enough.